backends/sycl-ref/ceed-sycl-ref.sycl.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other
+// CEED contributors. All Rights Reserved. See the top-level LICENSE and NOTICE
+// files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+#include "ceed-sycl-ref.hpp"
+
+#include <ceed/backend.h>
+#include <ceed/ceed.h>
+
+#include <string>
+#include <sycl/sycl.hpp>
+
+//------------------------------------------------------------------------------
+// SYCL preferred MemType
+//------------------------------------------------------------------------------
+static int CeedGetPreferredMemType_Sycl(CeedMemType *mem_type) {
+  *mem_type = CEED_MEM_DEVICE;
+  return CEED_ERROR_SUCCESS;
+}
+
+//------------------------------------------------------------------------------
+// Backend Init
+//------------------------------------------------------------------------------
+static int CeedInit_Sycl(const char *resource, Ceed ceed) {
+  char *resource_root;
+  CeedCallBackend(CeedSyclGetResourceRoot(ceed, resource, &resource_root));
+  if (std::strcmp(resource_root, "/gpu/sycl/ref") && std::strcmp(resource_root, "/cpu/sycl/ref")) {
+    // LCOV_EXCL_START
+    return CeedError(ceed, CEED_ERROR_BACKEND, "Sycl backend cannot use resource: %s", resource);
+    // LCOV_EXCL_STOP
+  }
```

Comment:
```suggestion
  CeedCheck(!std::strcmp(resource_root, "/gpu/sycl/ref") || !std::strcmp(resource_root, "/cpu/sycl/ref"), ceed, CEED_ERROR_BACKEND, "Sycl backend cannot use resource: %s", resource);
```

Reply:
Yeah, these `CeedCheck` replacements is what I was referring to earlier, not `CeedCallBackend`. I think that's being used everywhere (or at least the `ierr = ...; CEEDCHK` isn't being used.

Reply:
Yeah, I clued in belatedly to that. Looks like otherwise its all pretty consistent with the other backends